src/utils/VisualEditorProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { DocumentContext } from "../hooks/useDocument.tsx";
+import { EntityFieldsContext } from "../hooks/useEntityFields.tsx";
+import { TailwindConfig } from "./themeResolver.ts";
+import { YextSchemaField } from "../internal/types/entityFields.ts";
+import { TailwindConfigContext } from "../hooks/useTailwindConfig.tsx";
+
+type AllOrNothing<T extends Record<string, any>> =
+  | T
+  | Partial<Record<keyof T, never>>;
+
+type UniversalProps<T> = {
+  document: T;
+  children: React.ReactNode;
+};
+
+type EditorProps = {
+  entityFields: YextSchemaField[];
+  tailwindConfig: TailwindConfig;
+};
+
+type VisualEditorProviderProps<T> = UniversalProps<T> &
+  AllOrNothing<EditorProps>;
```

Comment:
So both entityFields and tailwindConfig are required or neither?

Reply:
Yes, this allows the VisualEditorProvider to be used for both the edit and location templates, since only the DocumentProvider was being used in the location template previously.